File: common/src/main/java/com/like/common/view/badgeview/BadgeViewHelper.java
package com.like.common.view.badgeview;

import android.content.Context;
import android.support.annotation.ColorInt;
import android.view.View;

/**
 * 未读消息数量显示的徽章视图，注意，target需要有ViewGroup的parent。
 */
public class BadgeViewHelper {
    private BadgeView badgeView;

    public BadgeViewHelper(Context context, View target) {
        badgeView = new BadgeView(context);
        badgeView.setTargetView(target);
    }

    public void setTextColor(@ColorInt int color) {
        badgeView.setBadgeTextColor(color);
    }

    /**
     * @param size 单位sp
     */
    public void setTextSize(int size) {
        badgeView.setBadgeTextSize(size);
    }

    public void setBackgroundColor(@ColorInt int color) {
        badgeView.setBadgeBackgroundColor(color);
    }

    public void setMessageCount(String messageCount) {
        badgeView.setBadgeCount(messageCount);
    }
}

File: common/src/main/java/com/like/common/view/badgeview/BadgeViewHelper.kt
package com.like.common.view.badgeview

import android.content.Context
import android.view.View
import androidx.annotation.ColorInt

class BadgeViewHelper(context: Context, target: View) {
    private val mBadgeView: BadgeView by lazy {
        BadgeView(context).apply {
            setTargetView(target)
        }
    }

    fun setMessageCount(messageCount: String, @ColorInt textColor: Int? = null, textSize: Int? = null, @ColorInt backgroundColor: Int? = null) {
        mBadgeView.setBadgeCount(messageCount)
        if (textColor != null) {
            mBadgeView.setBadgeTextColor(textColor)
        }
        if (textSize != null) {
            mBadgeView.setBadgeTextSize(textSize)
        }
        if (backgroundColor != null) {
            mBadgeView.setBadgeBackgroundColor(backgroundColor)
        }
    }

    fun getMessageCount(): String {
        return mBadgeView.text.toString()
    }
}